lib/migrations/createGenesisDao.ts

```diff
@@ -23,135 +29,130 @@ interface FounderSpec {
+
+    if (!founders || (founders.length === 0)) {
+      throw new Error(`no founders were given for the network: ${this.network}`);
+    }
+
+    let gasLimit = computeForgeOrgGasLimit(founders.length);
+    const maxGasLimit = await computeMaxGasLimit(this.web3);
+
+    gasLimit = Math.min(gasLimit, maxGasLimit);
+
+    console.log(`Forging ${orgName} to ${this.network}, gasLimit: ${gasLimit} and ${founders.length} founders...`);
 
-    await deployer.deploy(UController, { gas: gasAmount });
-    const universalControllerInst = await UController.deployed();
     /**
      * Create the Genesis DAO
      */
-    const tx = await daoCreatorInst.forgeOrg(
+    const txForgeOrg = await daoCreatorInst.forgeOrg(
       orgName,
       tokenName,
       tokenSymbol,
       founders.map((f: FounderSpec) => f.address),
-      founders.map((f: FounderSpec) => web3.toWei(f.tokens)),
-      founders.map((f: FounderSpec) => web3.toWei(f.reputation)),
+      founders.map((f: FounderSpec) => this.web3.toWei(f.tokens)),
+      founders.map((f: FounderSpec) => this.web3.toWei(f.reputation)),
       universalControllerInst.address,
-      web3.toWei(100000000), // token cap of one hundred million GEN, in Wei
-      { gas: gasAmount });
+      this.web3.toWei(100000000), // token cap of one hundred million GEN, in Wei
+      { gas: gasLimit });
 
-    // maximize chances that Avatar.at will succeed
-    if (network === "live") {
-      await UtilsInternal.sleep(60000);
+    /**
+     * save info for later steps
+     */
+    if (live) {
+      UtilsInternal.sleep(10000); // maximize chances this is ready
     }
 
-    const AvatarInst = await Avatar.at(tx.logs[0].args._avatar);
-    let genTokenAddress;
+    const avatarInst = await Avatar.at(txForgeOrg.logs[0].args._avatar);
+
+    /** for use by setSchemes */
+    return {
+      avatarAddress: avatarInst.address,
+      daoCreatorInst,
+      orgName,
+    } as ForgedDaoInfo;
+  }
+
+  public async setSchemes(forgedDaoInfo: ForgedDaoInfo): Promise<void> {
 
-    if ((network !== "live") && (network !== "kovan")) {
-      genTokenAddress = await AvatarInst.nativeToken();
-      console.log(`**** using native token for staking on network ${network} at: ${genTokenAddress}`);
-    } else {
-      // the global GEN token
-      genTokenAddress = "0x543Ff227F64Aa17eA132Bf9886cAb5DB55DCAddf";
-      console.log(`**** using global GEN token for staking on network ${network} at: ${genTokenAddress}`);
-    }
     /**
-     * The voting machine.  GenesisProtocol must be deployed as a scheme if it is
-     * to be used by schemes as a voting machine, which is what all of the
-     * Genesis schemes do.
+     * Truffle Solidity artifact wrappers
      */
-    await deployer.deploy(GenesisProtocol, genTokenAddress, { gas: gasAmount });
-    const genesisProtocolInst = await GenesisProtocol.deployed();
+    const ContributionReward = await Utils.requireContract("ContributionReward");
```

Comment:
The schemes list suppose to be passed from the user unless it choose the defaults.

Reply:
Separate PR to support creating arbitrary DAOs.